core/src/Kokkos_Atomics_Desul_Wrapper.hpp

```diff
@@ -0,0 +1,239 @@
+#ifndef KOKKOS_DESUL_ATOMICS_WRAPPER_HPP_
+#define KOKKOS_DESUL_ATOMICS_WRAPPER_HPP_
+#include <Kokkos_Macros.hpp>
+
+#ifdef KOKKOS_ENABLE_IMPL_DESUL_ATOMICS
+#include <desul/atomics.hpp>
+
+#include <impl/Kokkos_Atomic_Memory_Order.hpp>
+#include <impl/Kokkos_Volatile_Load.hpp>
+
+// clang-format off
+namespace Kokkos {
+
+// These functions don't have any use/test in unit tests ...
+// ==========================================================
+inline const char* atomic_query_version() { return "KOKKOS_DESUL_ATOMICS"; }
+
+#if defined(KOKKOS_COMPILER_GNU) && !defined(__PGIC__) && \
+    !defined(__CUDA_ARCH__)
+
+#define KOKKOS_NONTEMPORAL_PREFETCH_LOAD(addr) __builtin_prefetch(addr, 0, 0)
+#define KOKKOS_NONTEMPORAL_PREFETCH_STORE(addr) __builtin_prefetch(addr, 1, 0)
+
+#else
+
+#define KOKKOS_NONTEMPORAL_PREFETCH_LOAD(addr) ((void)0)
+#define KOKKOS_NONTEMPORAL_PREFETCH_STORE(addr) ((void)0)
+
+#endif
```

Comment:
Yes because Kokkos_Atomic is now:

```c++
#ifdef KOKKOS_IMPL_ENABLE_DESUL_ATOMICS
#include<Kokkos_Atomics_Desul_Wrapper.hpp>
#include<Kokkos_Atomics_Desul_Volatile_Wrapper.hpp>
#else

// All the stuff which was there before
#endif
```

And I strongly believe its much better to replicate this hear, which also reminds us that we have this internal shit, when sprinkling ifdefs throughout the old atomic file, and then later trying to see what we still use from that.